dask_ml/model_selection/_incremental.py

```diff
@@ -747,7 +845,8 @@ def __init__(
         self.tol = tol
         self.scores_per_fit = scores_per_fit
         self.max_iter = max_iter
-        super(IncrementalSearch, self).__init__(
+        self._all_models = set()
```

Comment:
Could you explain why this is needed?

Reply:
To keep models in memory, so the model with the best score (which may not be the last model used by `_fit`; e.g., it may have plateaued early). `_all_models` holds the IDs of all models and returns them from `_additional_calls`, so `_fit` doesn't remove those models.

The relevant lines are

https://github.com/dask/dask-ml/blob/76a6482d3a2fc87514caa670777e58d5dbb7a672/dask_ml/model_selection/_incremental.py#L211-L218

Reply:
Keeping all models in memory may not be ideal.  In cases where we've plateaued do we know that a higher score on the validation data actually implies a better model?  

Alternatively, maybe we just hold onto the best that we've seen so far rather than all models?  It would be nice to let things clear out.

Reply:
I think it'd suffice to hold onto one other model to protect against the mentioned case. I'll put in some work for that, hopefully tonight.

Reply:
The code holds onto fewer models now with 7de8f6a6061ed6b742a5179dd9308686ed0cd3a5.